pkg/envoy/rds/builder.go

```diff
@@ -0,0 +1,97 @@
+package rds
+
+import (
+	mapset "github.com/deckarep/golang-set"
+	xds_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	xds_discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/rds/route"
+	"github.com/openservicemesh/osm/pkg/trafficpolicy"
+)
+
+type routesBuilder struct {
+	routes []*xds_route.RouteConfiguration
+}
+
+func newRoutesBuilder() *routesBuilder {
+	return &routesBuilder {
+		routes: make([]*xds_route.RouteConfiguration, 0),
```

Comment:
instead of building here, we should be setting all inputs upfront, then call build at the end.

ie: add the TrustDomain as a field, trafficmatches, etc.